# 출력

In [3]:
print("Hello World", end=" ")
print("Test", end="\n")
print("Test2")

Hello World Test
Test2


n: New Line (개행)

우리가 end라는 어떤 파라미터 변수에 다른 값을 설정하면, print 함수가 문자열을 출력한 다음에 출력하는 끝 문자를 변경할 수가 있다.

기본형은 end='\n'이어서 어떤 문자열을 출력한 다음에, 기본적으로 개행을 진행한다.

# 자료형

자료형: 자료를 저장하는 형태

* 숫자형
* 문자열 자료형
* 리스트 자료형
* 튜플 자료형
* 딕셔너리 자료형
* 집합 자료형
* 불 자료형

## 숫자형

123

-123

0

1.2

-3.45


* 정수형: 소수점이 없는 숫자형

* 실수형: 소수점이 있는 숫자형

정수형: 123, -123, 0

실수형: 1.2, -3.45

* 사칙연산
    - ```+```
    - ```-```
    - ```*```
    - ```/```
  
  

* ```**``` 연산자

In [4]:
a = 3
b = 4

$a^{b}$

In [5]:
a ** b

81

## 나눗셈 후 나머지를 리턴하는 % 연산자

![percent](https://wikidocs.net/images/page/12/02_1_mod1.png)

In [6]:
7 % 3

1

In [7]:
3 % 7

3

### 나머지 연산자를 활용하는 경우

* 짝수인지/홀수인지 판별

In [8]:
a = 10

In [9]:
if a % 2 == 0:
    print("짝수")
else:
    print("홀수")

짝수


### 나눗셈 후 몫을 리턴하는 /, // 연산자

/ : 결과값이 몫을 반환하지만, 실수값이 나옴.

// : 결과값이 몫을 반환하나, 정수값이 나옴.

In [10]:
a = 36
b = 36
c = 36

print(a + b + c)

108


In [11]:
print((a + b + c) / 3)

36.0


In [12]:
print((a + b + c) // 3)

36


```shell
평균: 36
```

In [13]:
a = 37
b = 39
c = 321

In [14]:
print((a + b + c) / 3)

132.33333333333334


평균값: 132.333

(소수점 아래 셋째자리까지 출력)

위 문제를 해결하기 위해서는 우선 실수 값이 도출되어야 하므로, /을 사용하는 것이 적절.

In [15]:
print(f"평균: {(a + b + c) / 3:.3f}")

평균: 132.333


{}을 사용할 때는 내부에다가 콜론(:)을 적고 그 위에 점(.)을 찍은 다음에, 내가 출력하고 싶은 소수점 자릿수에 해당하는 숫자를 적은 다음에 마지막에 f문자만 붙여주면 됨.

위와 같이 소수점 아래 몇 째 자리까지 출력하는 문제가 나올 경우에는 위와 같은 {}를 사용하는 것 대신에 %f를 많이 사용함.

In [16]:
a = 10.12312312

In [17]:
print("a의 값: %f" % (a))

a의 값: 10.123123


실수를 출력할 때는 %f


소수점 아래 셋째자리까지 출력하고 싶을 때는 다음과 같이 작성함.

In [18]:
print("a의 값: %.3f" % (a))

a의 값: 10.123


문자열은 영어로 string

In [20]:
a = "Hello"

In [21]:
print("%s World" % (a))

Hello World


In [22]:
b = "World"

여러 개의 값을 보내고 싶을 때는 그만큼 여러개의 %f, %s 또는 %d를 작성하고, 그것의 개수에 맞게끔 % 뒤에 변수나 값들을 소괄호 내부에 전달해줘야 함.

In [23]:
print("%s %s" % (a, b))

Hello World


정수: Integer

숫자: xxxxxxx (decimal) : 십진법의, 소수

틀린 갯수: 7 (총 10번)

%d

In [24]:
a = 10

In [25]:
print("%d" % (a))

10


## 문자열 자료형


In [26]:
a = "Hello"

In [27]:
b = "Yerin's Hamster"

In [28]:
c = 'World'

In [29]:
d = "Yerin's Hamster"

In [ ]:
e = "Yerin's Hamster said "I am very happy""

In [1]:
f = "Yerin's  Hamster said \"I am very happy\""

In [2]:
f

'Yerin\'s  Hamster said "I am very happy"'

In [3]:
g = "Python"
h = "Language"

In [4]:
g + h

'PythonLanguage'

In [5]:
g + " " + h

'Python Language'

=====================================

Hello World

=====================================

In [6]:
len("=====================================")

37

In [8]:
print("=" * 37)
print()
print("Hello World")
print()
print("=" * 37)


Hello World



In [10]:
a = "Life is too short and you need Python"

문자열(String)은 다음의 특성을 가진다.

Immutable: 변할 수 없음.

### Indexing

그 전체 문자열 중에서 하나의 문자만을 취하고 싶을 때

컴퓨터는 항상 첫 번째 숫자가 1이 아닌, 0이다.

In [11]:
a[0]

'L'

In [12]:
a[0] = "T"

TypeError: 'str' object does not support item assignment

### Slicing

값을 자를 때, 전체 문자열에서 부분 문자열을 취하고 싶을 때 사용한다.

In [13]:
a = "T" + a[1:]

In [14]:
a

'Tife is too short and you need Python'

원래는 문자열의 값을 바꾸는 연산이 허용되지 않지만, 위와 같이 슬라이싱(Slicing)을 이용하면 문자열의 값을 바꿀 수가 있다.

In [15]:
a[:]

'Tife is too short and you need Python'

콜론(:)의 앞이 비어있으면, 처음부터라는 의미이고, 뒤가 비어있으면, 끝까지라는 의미입니다.

둘다 비어있으면, 처음부터 끝까지라는 의미입니다.

둘다 비어있는 경우에는 다음의 두 구문이 서로 동일합니다.

```python
a

a[:]
```

In [16]:
print("%s" % a)

Tife is too short and you need Python


In [17]:
b = "%s" % (a)

In [19]:
b = "%s" % ("Tife is too short and you need Python")

In [20]:
b = "Tife is too short and you need Python"

우리가 위에서 봤던 여러가지의 자료형(값의 형태)들은 모두 값이다.

그러므로 위의 "Tife is too short and you need Python"은 문자열이므로 값이다.

In [18]:
b

'Tife is too short and you need Python'

항상 명심해야할 사항,,,

변수의 경우에는, 

```python
a = "Hello"

a = 50
```

"Hello"와 50은 값이다.

```python
a = b

a = b + c
```

```a = b```의 경우에는 등호의 오른쪽은 값이 아닌, 변수의 이름,,,

이러한 경우에는 일단 그 변수에 들어가 있는 값이 나오고, 그 값을 다시 왼쪽에 있는 a라는 변수에 넣는 것임.

```python
b = 10
```

1. 우리가 작성한 구문

```python
a = b
```

2. 파이썬이 위의 문장을 이해하는 순서

```python
a = 10
```

3. 그러므로 최종적으로 a라는 변수에 들어가는 값은 10이다. (왜냐하면 등호의 오른쪽에 있는 값이 항상 왼쪽에 있는 변수에 들어가기 때문에)

```python
c = 20
```



4. 우리가 작성한 구문


```python
a = b + c
```


5. 파이썬이 위의 문장을 이해하는 순서

```python
a = 10 + c
```


```python
a = 10 + 20
```


```python
a = 30
```

6. 그러므로 최종적으로 a라는 변수에 들어가는 값은 30이다.





## 리스트 자료형

리스트는 자료형이므로 값이다.

리스트에 값을 넣을 때는 대괄호(Square bracket)를 사용한다.

In [21]:
odd = [1, 3, 5, 7, 9]

In [22]:
print(odd)

[1, 3, 5, 7, 9]


리스트는 이전의 자료형들과는 다르게 여러가지의 값들을 저장할 수 있는 또 다른 값이다. (값들의 값)

리스트는 값들을 담은 약간 보따리(값) 같은 느낌

In [23]:
odd = [1, "3", 5, 7, 9]

In [24]:
print(odd)

[1, '3', 5, 7, 9]


우리가 리스트 안에는 여러가지의 값들을 넣을 수가 있습니다. (하나의 값을 넣으라는 규칙이 있지는 않아요,,,)

### 인덱싱(Indexing)

In [25]:
odd[0]

1

### 슬라이싱(Slicing)

In [26]:
odd[1:]

['3', 5, 7, 9]

문자열에서 그러했던 것처럼 리스트에서도 동일한 연산들(인덱싱(Indexing), 슬라이싱(Slicing))이 적용된다. 